lib/build/dispatcher/dispatcher.js

```diff
@@ -233,26 +215,38 @@ class Dispatcher {
    * @param {string} mode - The mode of build target.
    * @param {string} entry - The entry point for the build.
    * @param {boolean} useNodePolyfills - The flag to indicates polyfills use.
+   * @param {boolean} useOwnWorker - This flag indicates that the constructed code inserts its own worker expression,
+   * such as addEventListener("fetch") or similar, without the need to inject a provider.
    */
-  constructor(preset, mode, entry, useNodePolyfills) {
+  constructor(preset, mode, entry, useNodePolyfills, useOwnWorker) {
     this.preset = preset;
     this.mode = mode;
     this.entry = entry;
     this.useNodePolyfills = useNodePolyfills;
+    this.useOwnWorker = useOwnWorker;
     this.buildId = generateTimestamp();
   }
 
   /**
    * Run the build process.
    */
   run = async () => {
-    await checkNodeModulesDir();
+    // Check install depenedencies
```

Comment:
We must reactivate `node_modules` check. Some projects maybe don't have dependencies or a package.json (simple functions).